core/src/main/java/org/mule/runtime/core/internal/execution/FlowProcessMediator.java

```diff
@@ -193,21 +193,15 @@ private void dispatch(PhaseContext ctx) throws Exception {
 
                                  @Override
                                  public void complete(Either<SourcePolicyFailureResult, SourcePolicySuccessResult> value) {
-                                   //TODO: Remove this if when MULE-17386 is fixed
-                                   if (ctx.result == null) {
-                                     ctx.result = value;
-                                     dispatchResponse(ctx);
-                                   }
+                                   ctx.result = value;
+                                   dispatchResponse(ctx);
                                  }
 
                                  @Override
                                  public void error(Throwable e) {
-                                   //TODO: Remove this if when MULE-17386 is fixed
-                                   if (ctx.result == null) {
```

Comment:
when I added this workaround I was trying the proxy performance test, which doesn't use any policies. Are you sure that the fix in `CommonSourcePolicy` is enough to be able to resolve this?